src/shogun/mathematics/Random.h

```diff
@@ -31,7 +32,7 @@ namespace shogun
 	 * number generator.
 	 *
 	 * */
-	class CRandom : public CSGObject
+	class CRandom
```

Comment:
Because we want to call m_rng->set_set_seed() in CSObject, so it's impossible to do that if CRandom still inherit from CSGObject.